rxandroidble/src/test/groovy/com/polidea/rxandroidble/internal/operations/OperationCharacteristicLongWriteTest.groovy

```diff
@@ -203,50 +203,164 @@ public class OperationCharacteristicLongWriteTest extends Specification {
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> { BluetoothGattCharacteristic characteristic ->
+            onCharacteristicWriteSubject.onNext(new ByteAssociation<UUID>(characteristic.getUuid(), [] as byte[]))
+            true
+        }
+
+        then:
+        1 * mockCharacteristic.setValue([0x3, 0x3] as byte[]) >> true
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> { BluetoothGattCharacteristic characteristic ->
+            onCharacteristicWriteSubject.onNext(new ByteAssociation<UUID>(characteristic.getUuid(), [] as byte[]))
+            true
+        }
+
+        testSubscriber.assertValueEquals([0x1, 0x1, 0x2, 0x2, 0x3, 0x3] as byte[])
+        testSubscriber.assertCompleted()
+    }
+
+    def "attempt to rewrite the failed batch if the strategy has emitted the LongWriteFailure - mid batch"() {
+        given:
+        this.writeOperationRetryStrategy = givenWillRetryWriteOperation()
+        this.writeOperationAckStrategy = new ImmediateSerializedBatchAckStrategy()
+        prepareObjectUnderTest(2, [0x1, 0x1, 0x2, 0x2, 0x3, 0x3] as byte[])
+
+        when:
+        objectUnderTest.run(mockQueueReleaseInterface).subscribe(testSubscriber)
+
+        then:
+        1 * mockCharacteristic.setValue([0x1, 0x1] as byte[]) >> true
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> { BluetoothGattCharacteristic characteristic ->
+            onCharacteristicWriteSubject.onNext(new ByteAssociation<UUID>(characteristic.getUuid(), [] as byte[]))
+            true
+        }
+
+        then:
+        1 * mockCharacteristic.setValue([0x2, 0x2] as byte[]) >> true
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> false
+
+        when:
+        writeOperationRetryStrategy.triggerRetry()
+
+        then:
+        1 * mockCharacteristic.setValue([0x2, 0x2] as byte[]) >> true
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> { BluetoothGattCharacteristic characteristic ->
+            onCharacteristicWriteSubject.onNext(new ByteAssociation<UUID>(characteristic.getUuid(), [] as byte[]))
+            true
+        }
+
+        then:
+        1 * mockCharacteristic.setValue([0x3, 0x3] as byte[]) >> true
+        1 * mockGatt.writeCharacteristic(mockCharacteristic) >> { BluetoothGattCharacteristic characteristic ->
+            onCharacteristicWriteSubject.onNext(new ByteAssociation<UUID>(characteristic.getUuid(), [] as byte[]))
+            true
+        }
+
+        testSubscriber.assertValueEquals([0x1, 0x1, 0x2, 0x2, 0x3, 0x3] as byte[])
+        testSubscriber.assertCompleted()
+    }
+
+    def "attempt to rewrite the failed batch if the strategy has emitted the LongWriteFailure - last batch"() {
+        given:
+        this.writeOperationRetryStrategy = givenWillRetryWriteOperation()
+        this.writeOperationAckStrategy = new ImmediateSerializedBatchAckStrategy()
+        prepareObjectUnderTest(2, [0x1, 0x1, 0x2, 0x2, 0x3, 0x3] as byte[])
```

Comment:
Could you make the last batch to have an uneven number of bytes? I think we can have a wrong `ByteBuffer` positioning on the last batch if is not full